flang/lib/Lower/OpenMP/OpenMP.cpp

```diff
@@ -2034,6 +2034,52 @@ static void genStandaloneDo(lower::AbstractConverter &converter,
                 llvm::omp::Directive::OMPD_do, dsp);
 }
 
+static void genLoopClauses(
+    lower::AbstractConverter &converter, semantics::SemanticsContext &semaCtx,
+    const List<Clause> &clauses, mlir::Location loc,
+    mlir::omp::LoopOperands &clauseOps,
+    llvm::SmallVectorImpl<const semantics::Symbol *> &reductionSyms) {
+  ClauseProcessor cp(converter, semaCtx, clauses);
+  cp.processOrder(clauseOps);
+  cp.processReduction(loc, clauseOps, reductionSyms);
+  cp.processTODO<clause::Bind, clause::Lastprivate>(
```

Comment:
Is `lastprivate` not already handled by the call in `createBodyOfOp()` to `DataSharingProcessor::processStep2()`?

Reply:
Not out-of-the-box. We hit this TODO: https://github.com/llvm/llvm-project/blob/main/flang/lib/Lower/OpenMP/DataSharingProcessor.cpp#L260. If you don't mind, I prefer to handle this in its own PR to focus on testing `lastprivate`.